### ISSUE TITLE
Android custom Markers with children content have sporadic flashing/glitches (see example video)

### ISSUE THREAD

**Issue by trooperandz** [react-native-maps/react-native-maps#4988]
### Summary

Beginning with `react-native-maps` version 1.9.0, map Markers composed of children content (described [in the documentation](https://github.com/react-native-maps/react-native-maps/blob/master/docs/marker.md) as Children Content) are showing sporadic flashes/glitching on Android. This does not occur with versions 1.8.4 and below.

In this particular case, the Markers are being rendered from an iterable list via `map`, and they are of SVG format.  The particular screen in question is only rendering one time (there are not multiple renders taking place).

The newest versions of `react-native-maps` were tried as well, with the same flashing occurring (1.10.0, 1.10.3).

### Reproducible sample code

```TSX
import * as React from 'react';
import { View } from 'react-native';
import MapView, { Marker } from 'react-native-maps';
import PinIcon from 'assets/icon-pin.svg';

<MapView>
  {markers.map((marker) => {
    return (
      <Marker
        identifier={`id-${marker.id}`}
        key={`key-id-${marker.id}`}
      >
        <View style={styles.pin}>
          <PinIcon
            fill={color}
            height={imageDimensions}
            width={imageDimensions}
          />
        </View>
      </Marker>
    );
  })}
</MapView>

const styles = StyleSheet.create({
  pin: {
    backgroundColor: 'rgba(0,0,0,0.1)',
    borderRadius: 13,
    height: 25,
    width: 25,
  },
})
```

### Steps to reproduce

Create a map, and render a `MapView` which includes Markers with children content rendered from an iterable list.  Observe the flashing/glitching of the markers.

### Expected result

Expect no flashing/glitches in the Marker content to occur.

### Actual result

See the flashing/glitches below:

https://github.com/react-native-maps/react-native-maps/assets/14365371/033aeb27-b179-488c-a751-6efe3a2817eb

### React Native Maps Version

1.9.0

### What platforms are you seeing the problem on?

Android

### React Native Version

0.73.4

### What version of Expo are you using?

Not using Expo

### Device(s)

Samsung Galaxy S21 (Android 13)

**Comment by pauladapptor** [comment]
Ran into a similar issue, if you remove the image as a child and set it as a prop, this should solve the flickering. Below is just some sudo code of your example, you may have to tweak it for your asset.

```typescript
<MapView>
  {markers.map((marker) => {
    return (
      <Marker
        identifier={`id-${marker.id}`}
        key={`key-id-${marker.id}`} 
        image={PinIcon}/>
    );
  })}
</MapView>
```

**Comment by Jin-seop** [comment]
i solved this
```
const [tracksView, setTracksView] = useState(true); //add 

useEffect(() => { //add
  if(Platform.OS === 'android'){
    setTimeout(()=>{
      setTracksView(false);
    },100)
  }
},[])

<MapView>
  {markers.map((marker) => {
    return (
      <Marker
        identifier={`id-${marker.id}`}
        key={`key-id-${marker.id}`}
        tracksViewChanges={tracksView} //add
      >
        <View style={styles.pin}>
          <PinIcon
            fill={color}
            height={imageDimensions}
            width={imageDimensions}
          />
        </View>
      </Marker>
    );
  })}
</MapView>
```

**Comment by chanphiromsok** [comment]
try to custom marker and set trackView to false
```js
import React, { ReactNode, useEffect, useState } from 'react';
import { StyleProp, ViewStyle } from 'react-native';
import { Marker, MarkerAnimated, MapMarkerProps } from 'react-native-maps';

interface Props extends MapMarkerProps {
  animate?: boolean;
  children: ReactNode;
  renderChangeTime?: number;
  style?: StyleProp<ViewStyle>;
}
export default function OptimizedMarker({
  animate = false,
  children,
  renderChangeTime = 2000,
  ...props
}: Props) {
  const [tracksViewChanges, setViewChanges] = useState(true);

  useEffect(() => {
    let timerId:NodeJS.Timeout

    if (tracksViewChanges) {
      timerId = setTimeout(() => {
        setViewChanges(false);
      }, renderChangeTime);
    }

    return () => {
      clearTimeout(timerId);
    };
  }, []);

  if (animate) {
    return (
      <MarkerAnimated {...props} tracksViewChanges={tracksViewChanges}>
        {children}
      </MarkerAnimated>
    );
  }

  return (
    <Marker tracksViewChanges={tracksViewChanges} {...props}>
      {children}
    </Marker>
  );
}
```
and try use png instead of svg

**Comment by andresvega-sourcetoad** [comment]
The issue when updated to 1.10.3 and the same behavior on 1.11.3 trackViewChanges on false doesn't solve the problem for me.

**Comment by jeffinhk** [comment]
> The issue ocurss when updated to 1.10.3 and the same behavior on 1.11.3 trackViewChanges on false doesn't solve the problem for me.

In my case trackViewChanges on false solves the issue temporarily but instead stops the Marker to render when entering the app after closing. So now the custom marker performance is either flickering or not render at all

**Comment by salah-ghanim** [comment]
@jeffinhk in our application the usage of onMapReady on android solves the issue ... 

@trooperandz have you tried using both onMapReady and tracksViewChanges to be false, it should solve the problem completely

**Comment by nahuelprinsich** [comment]
I'm facing the same issue. If I Take the last advice using tracksViewChanges with onMapReady the flickering stops but some markers get stuck with an image from another marker.
Also tried another advices but nothing seems to work, for example use googleRender='LEGACY' or use a timer to set tracksViewChanges to false after a second.

**Comment by trooperandz** [comment]
> @jeffinhk in our application the usage of onMapReady on android solves the issue ...
> 
> @trooperandz have you tried using both onMapReady and tracksViewChanges to be false, it should solve the problem completely

@salah-ghanim yes I tried both of these approaches; I need `tracksViewChanges` to be true, because my pins react to presses (change to larger pins to show that they have been pressed by the user).

**Comment by nahuelprinsich** [comment]
I fixed the problem using googleRenderer={'LEGACY'}

**Comment by trooperandz** [comment]
> I fixed the problem using googleRenderer={'LEGACY'}

Interesting; this does not fix it for me unfortunately.

**Comment by ryskin** [comment]
It didn't work 

> try to custom marker and set trackView to false
> 
> ```js
> import React, { ReactNode, useEffect, useState } from 'react';
> import { StyleProp, ViewStyle } from 'react-native';
> import { Marker, MarkerAnimated, MapMarkerProps } from 'react-native-maps';
> 
> interface Props extends MapMarkerProps {
>   animate?: boolean;
>   children: ReactNode;
>   renderChangeTime?: number;
>   style?: StyleProp<ViewStyle>;
> }
> export default function OptimizedMarker({
>   animate = false,
>   children,
>   renderChangeTime = 2000,
>   ...props
> }: Props) {
>   const [tracksViewChanges, setViewChanges] = useState(true);
> 
>   useEffect(() => {
>     let timerId:NodeJS.Timeout
> 
>     if (tracksViewChanges) {
>       timerId = setTimeout(() => {
>         setViewChanges(false);
>       }, renderChangeTime);
>     }
> 
>     return () => {
>       clearTimeout(timerId);
>     };
>   }, []);
> 
>   if (animate) {
>     return (
>       <MarkerAnimated {...props} tracksViewChanges={tracksViewChanges}>
>         {children}
>       </MarkerAnimated>
>     );
>   }
> 
>   return (
>     <Marker tracksViewChanges={tracksViewChanges} {...props}>
>       {children}
>     </Marker>
>   );
> }
> ```
> 
> and try use png instead of svg

this works!

**Comment by oleg1994** [comment]
Still happening to me, is there anyway to fix this ?

**Comment by yperdimis** [comment]
Please update the react native maps library so we can select the legacy renderer in react native MapView. Test the library by putting a view inside marker with an external image to see how horrible it flickers. The use of prop trackViewChanges={true} and set to {false} after some time makes mess on code.

**Comment by aflatoon2874** [comment]
I am also facing the same issue. I am on version 1.14.0. Is this issue being worked at? Or any other solution? Please help.

**Comment by DavidAmyot** [comment]
+1

**Comment by trooperandz** [comment]
Adding `googleRenderer="LEGACY"` to _all_ map instances fixed the issue for us.  It turns out that when I had tested this previously, we learned that if you have multiple `<MapView>` instances in your app, the underlying google renderer seems to initialize only one time.  So if the initial map you render does not have `googleRenderer="LEGACY"`, then a subsequently-rendered `<MapView>` with `googleRenderer="LEGACY"` will actually not use the legacy renderer (it won't take effect, because the renderer engine only initializes the first time it seems).  

In our case, adding this property to the very first map that renders fixed all of the others, regardless of if the other <MapView> instances had the `googleRenderer` prop or not.  For sanity and extra caution, we added it to all map instances.
